compiler/semstmts.nim

```diff
@@ -1916,9 +1918,9 @@ proc semProcAux(c: PContext, n: PNode, kind: TSymKind,
 
   if not hasProto and sfGenSym notin s.flags: #and not isAnon:
     if s.kind in OverloadableSyms:
-      addInterfaceOverloadableSymAt(c, delcarationScope, s)
+      addInterfaceOverloadableSymAt(c, declarationScope, s)
     else:
-      addInterfaceDeclAt(c, delcarationScope, s)
+      addInterfaceDeclAt(c, declarationScope, s)
```

Comment:
The addition to the interface should happen after the potential reset of the scope if a prototype is found, around line 1955.

Reply:
Also document the heck out of why that's important.